sites/public/layouts/eligibility.tsx

```diff
@@ -2,14 +2,54 @@ import Layout from "./application"
 import styles from "./Eligibility.module.scss"
 import { ApplicationTimeout } from "../src/forms/applications/ApplicationTimeout"
 import React, { useContext } from "react"
-import { ProgressNav, t } from "@bloom-housing/ui-components"
+import {
+  AppearanceStyleType,
+  Button,
+  Form,
+  FormCard,
+  ProgressNav,
+  t,
+} from "@bloom-housing/ui-components"
 import { EligibilityContext } from "../lib/EligibilityContext"
 import { ELIGIBILITY_SECTIONS } from "../lib/constants"
 import { eligibilityRoute } from "../lib/helpers"
+import { UseFormMethods } from "react-hook-form"
+import { useRouter } from "next/router"
+import { getFilterUrlLink } from "../lib/filterUrlLink"
 
-const EligibilityLayout = (props) => {
+export interface EligibilityLayoutProps {
+  title: string
+  currentPage: number
+  children: any
+  onFinishStep?: (data: any) => void
+  formMethods: UseFormMethods
+}
+
+const EligibilityLayout = (props: EligibilityLayoutProps) => {
+  const router = useRouter()
   const { eligibilityRequirements } = useContext(EligibilityContext)
 
+  const handleSubmit = props.formMethods.handleSubmit
+
+  const onBack = async (data) => {
+    props.onFinishStep && props.onFinishStep(data)
+    await router.push(eligibilityRoute(props.currentPage - 1))
+  }
+  const onNext = async (data) => {
+    props.onFinishStep && props.onFinishStep(data)
+    await router.push(eligibilityRoute(props.currentPage + 1))
+  }
+  const onFinish = async (data) => {
+    props.onFinishStep && props.onFinishStep(data)
+    await router.push(getFilterUrlLink(eligibilityRequirements))
```

Comment:
When I try clicking on "See results now" it doesn't seem to be applying the filters? Is this working for you?

Reply:
It seems this is just broken generally... Looking at `filterUrlLink.tsx`, I'm not sure what the logic is supposed to be for applying eligibility form data to a filter. :disappointed:

Reply:
I filed a bug for this: https://github.com/CityOfDetroit/bloom/issues/892